app/services/pages/loader.rb

```diff
@@ -0,0 +1,23 @@
+module Pages
+  class Loader
+    def self.load
+      Pages::Manager.add_page({ :controller => :smart_proxies, :action => :show }, "smart_proxies/show") do |page|
+        page.add_extension_point :tabbed_menu
+
+        page.add_pagelet(:name => "Overview",
+                         :mountpoint => :tabbed_menu,
+                         :priority => 50,
+                         :partial => "smart_proxies/tabs/overview") do |overview|
+          overview.add_extension_point :foreman_details
+          overview.add_pagelet :name => "Details", :mountpoint => :foreman_details, :partial => 'smart_proxies/widgets/details'
+
+          #Katello will attach itself on this with info from capsule
+          overview.add_extension_point :katello_details
+        end
+        page.add_pagelet(:name => "Services", :mountpoint => :tabbed_menu, :partial => "smart_proxies/tabs/services") do |services|
+          services.add_extension_point :services
+        end
+      end
+    end
+  end
+end
```

Comment:
I think we can completely get rid of the pre-initialization.

Let's suppose that if you try to render an extension point that has no pagelets, you simply return an empty array. Therefore nothing will be rendered, as simple as that. Remove also the early errors when things are not present.

The only benefit I see in having this is to be able to generate list of extensions, but is it worth the effort?